### PR TITLE
test: Add tests for client error console and card kitchensink

### PR DESCRIPTION
**Pull request by karangattu**
Fixes https://github.com/posit-dev/py-shiny/issues/1251

- [x] Add resconnect json file before merge
- [x] Fix check error in CI - https://github.com/posit-dev/py-shiny/actions/runs/8510801310/job/23309139604

### CODE REVIEW

File: tests/playwright/deploys/shiny-client-error/test_shiny_client_error.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+@skip_if_not_chrome
+def test_shiny_client_error(page: Page, app_url: str) -> None:
+    page.goto(app_url)
+
+    assert page.locator("#same_id").count() == 2
+    shiny_error_message = page.query_selector("shiny-error-message")
+
+    # show the client error message only for local apps
+    if "127.0.0.1:" in app_url:
```
Comment (karangattu) [inline tests/playwright/deploys/shiny-client-error/test_shiny_client_error.py:15]: If `http` not in app_url

File: tests/playwright/controls.py [hunk 2]
```diff
@@ -351,22 +351,54 @@ def expect_label(
...
+        """
+        Expect the width attribute of an element to have a specific value.
+
+        Parameters
+        ----------
+            value
+                The expected value of the width attribute.
+            timeout
+                The maximum time to wait for the expectation to be fulfilled.
+                Defaults to None.
```
Comment (schloerke) [inline tests/playwright/controls.py:374]: Left align parameter names. Indent parameter description. (Repeat for remaining parameter sections)

```suggestion
        value
            The expected value of the width attribute.
        timeout
            The maximum time to wait for the expectation to be fulfilled.
            Defaults to None.
```

File: tests/playwright/controls.py [hunk 3]
```diff
@@ -351,22 +351,54 @@ def expect_label(
...
+        """
         expect_attr(self.loc, "width", value=value, timeout=timeout)
 
 
 class _WidthContainerM:
+    """
+    A class representing the width container mixin.
```
Comment (schloerke) [inline tests/playwright/controls.py:381]:
```suggestion
    A mixin class representing the container's width.
```

File: tests/playwright/controls.py [hunk 4]
```diff
@@ -351,22 +351,54 @@ def expect_label(
 
 
 class _WidthLocM:
+    """
+    A class representing the width location mixin.
```
Comment (schloerke) [inline tests/playwright/controls.py:355]:
```suggestion
    A mixin class representing the `loc`'s width.
```

File: tests/playwright/controls.py [hunk 5]
```diff
@@ -2854,6 +2886,27 @@ def expect_full_screen_available(
 
 
 class Card(_WidthLocM, _CardFooterM, _CardBodyM, _CardFullScreenM, _InputWithContainer):
+    """
+    Card control for shiny.ui.card - https://shiny.posit.co/py/api/core/ui.card.html
```
Comment (schloerke) [inline tests/playwright/controls.py:2890]:
```suggestion
    Card control for :func:`~shiny.ui.card`
```

File: tests/playwright/controls.py [hunk 6]
```diff
@@ -2854,6 +2886,27 @@ def expect_full_screen_available(
...
+    """
+
+    loc: Locator
+    """
+    Locator for the card's value
```
Comment (schloerke) [inline tests/playwright/controls.py:2895]: Make class name within a code block

```suggestion
    `Locator` for the card's value
```

File: tests/playwright/controls.py [hunk 7]
```diff
@@ -2854,6 +2886,27 @@ def expect_full_screen_available(
...
+    """
+    Locator for the card title
+    """
+    loc_footer: Locator
+    """
+    Locator for the card footer
+    """
+    loc_body: Locator
+    """
+    Locator for the card body
```
Comment (schloerke) [inline tests/playwright/controls.py:2907]:
```suggestion
    `Locator` for the card title
    """
    loc_footer: Locator
    """
    `Locator` for the card footer
    """
    loc_body: Locator
    """
    `Locator` for the card body
```

File: tests/playwright/controls.py [hunk 8]
```diff
@@ -2909,14 +2982,87 @@ def expect_header(
...
+        Parameters
+        ----------
+        value
+            The expected height value.
+        timeout
+            The maximum time to wait for the expectation to pass. Defaults to None.
+        """
         expect_to_have_style(self.loc_container, "height", value, timeout=timeout)
 
+    def expect_full_screen_available(
```
Comment (schloerke) [inline tests/playwright/controls.py:3023]: Double check if we can remove the method from valuebox as it is a card under the hood. If it is not a card, leave as is.

**Review by schloerke — CHANGES_REQUESTED** [review]

File: tests/playwright/controls.py [hunk 9]
```diff
@@ -2693,31 +2725,48 @@ def expect_full_screen(
             int(open), timeout=timeout
         )
 
+    def expect_full_screen_enabled(
```
Comment (schloerke) [inline tests/playwright/controls.py:2728]: You were right. Reverting name change

```suggestion
    def expect_full_screen_available(
```

File: tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py [hunk 10]
```diff
@@ -57,13 +57,13 @@ def test_valuebox(page: Page, local_app: ShinyAppProc) -> None:
     assert get_value_tag_name(value_box1) == "h1"
     assert get_value_box_bg_color(value_box1) == "rgb(193, 0, 0)"
     assert get_value_box_fg_color(value_box1) == "rgb(255, 255, 255)"
-    value_box1.expect_full_screen_available(True)
-    value_box1.expect_full_screen(False)
+    value_box1.expect_full_screen_enabled(True)
```
Comment (schloerke) [inline tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py:60]:
```suggestion
    value_box1.expect_full_screen_available(True)
```

File: tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py [hunk 11]
```diff
@@ -74,7 +74,7 @@ def test_valuebox(page: Page, local_app: ShinyAppProc) -> None:
     value_box2.expect_value("Showcase left center")
     assert get_value_tag_name(value_box2) == "h5"
     assert get_value_box_bg_color(value_box2) == "rgb(0, 123, 194)"
-    value_box2.expect_full_screen_available(False)
+    value_box2.expect_full_screen_enabled(False)
```
Comment (schloerke) [inline tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py:77]:
```suggestion
    value_box2.expect_full_screen_available(False)
```

File: tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py [hunk 12]
```diff
@@ -84,7 +84,7 @@ def test_valuebox(page: Page, local_app: ShinyAppProc) -> None:
     value_box3.expect_title("No theme w/ Fullscreen")
     value_box3.expect_value("Showcase bottom")
     assert get_value_tag_name(value_box3) == "h3"
-    value_box3.expect_full_screen_available(True)
+    value_box3.expect_full_screen_enabled(True)
```
Comment (schloerke) [inline tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py:87]:
```suggestion
    value_box3.expect_full_screen_available(True)
```

File: tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py [hunk 13]
```diff
@@ -93,7 +93,7 @@ def test_valuebox(page: Page, local_app: ShinyAppProc) -> None:
     value_box4.expect_value("No theme - only defaults")
     assert get_title_tag_name(value_box4) == "p"
     assert get_title_tag_name(value_box4) == "p"
-    value_box4.expect_full_screen_available(False)
+    value_box4.expect_full_screen_enabled(False)
```
Comment (schloerke) [inline tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py:96]:
```suggestion
    value_box4.expect_full_screen_available(False)
```

File: tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py [hunk 14]
```diff
@@ -102,6 +102,6 @@ def test_valuebox(page: Page, local_app: ShinyAppProc) -> None:
     value_box5.expect_title("No showcase w/ showcase layout")
     value_box5.expect_value("Red text - fill is False")
     assert get_value_tag_name(value_box5) == "p"
-    value_box5.expect_full_screen_available(False)
+    value_box5.expect_full_screen_enabled(False)
```
Comment (schloerke) [inline tests/playwright/shiny/components/value_box/kitchensink/test_valuebox_ks.py:105]:
```suggestion
    value_box5.expect_full_screen_available(False)
```

File: tests/playwright/shiny/experimental/card/kitchensink/test_card_ks.py [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+    card.expect_min_height(None)
+    card.expect_height(None)
+    card.expect_header(None)
+    card.expect_footer(None)
+    card.expect_body(
+        ["\nThis is the body without a header of a footer - No Fullscreen\n"]
+    )
+    assert get_body_tag_name(card) == "p"
+    card.expect_full_screen_open(False)
+    card.expect_full_screen_enabled(False)
```
Comment (schloerke) [inline tests/playwright/shiny/experimental/card/kitchensink/test_card_ks.py:53]:
```suggestion
    card.expect_full_screen_available(False)
```

File: tests/playwright/shiny/experimental/card/kitchensink/test_card_ks.py [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+
+    card = Card(page, "card3")
+    card.expect_max_height("500px")
+    card.expect_min_height("200px")
+    card.expect_header("Fill is False. Fullscreen is False")
+    card.expect_footer(None)
+    card.expect_body(["Max height and min height are set."])
+    assert get_body_tag_name(card) == "h3"
+    card.expect_full_screen_enabled(False)
```
Comment (schloerke) [inline tests/playwright/shiny/experimental/card/kitchensink/test_card_ks.py:62]:
```suggestion
    card.expect_full_screen_available(False)
```

**Review by schloerke — APPROVED** [review]

**Pull request merged**